packages/cdktf/lib/terraform-hcl-module.ts

```diff
@@ -0,0 +1,81 @@
+import { Construct } from "constructs";
+import { TerraformModuleOptions, TerraformModule } from "./terraform-module";
+import { TerraformProvider } from "./terraform-provider";
+
+export interface TerraformHclModuleOptions extends TerraformModuleOptions {
+    readonly providers?: (TerraformProvider | TerraformModuleProvider)[];
+    readonly variables?: {[key: string]: any};
+}
+
+export interface TerraformModuleProvider {
```

Comment:
I'm not completely sold on this interface, but it implementation a bit easier while still not forcing a module alias.